src/make/Processor.cpp

```diff
@@ -742,46 +747,87 @@ Processor::_BuildCommand(data::RuleActionsCall* actionsCall)
 	// and "<" and ">")
 	data::VariableDomain builtInVariables;
 
-	// TODO: Support RuleActions::UPDATED
-	// TODO: Support RuleActions::EXISTING
-	auto setBoundTargets =
-		[this](StringList& boundTargets, const data::TargetList& targets) {
+	const std::uint32_t flags =
+		actionsCall->Actions()->Flags() & data::RuleActions::FLAG_MASK;
+	const bool isExistingAction = flags & data::RuleActions::EXISTING;
+	const bool isUpdatedAction = flags & data::RuleActions::UPDATED;
+	const bool isTrimmedSources = isExistingAction || isUpdatedAction;
+	const Target* primaryTarget = *actionsCall->Targets().begin();
+	const MakeTarget* primaryMakeTarget = _GetMakeTarget(primaryTarget, true);
+
+	const auto setBoundTargets =
```

Comment:
I'm wondering if this lambda has now become long enough to warrant it's own private method after all... :-)

Reply:
Yes, I am just about to do that haha